views/components/ConfigurationsList/index.jsx

```diff
@@ -0,0 +1,851 @@
+        inputProps: {
+          className: props.selectProps.classes.input,
+          inputRef: props.innerRef,
+          children: props.children,
+          ...props.innerProps,
+        },
+      }}
+      {...props.selectProps.textFieldProps}
+    />
+  )
+}
+
+function Option(props) {
+  return (
+    <MenuItem
+      buttonRef={props.innerRef}
+      selected={props.isFocused}
+      component="div"
+      style={{
+        fontWeight: props.isSelected ? 500 : 400,
+      }}
+      {...props.innerProps}
+    >
+      {props.children}
+    </MenuItem>
+  )
+}
+
+function Placeholder(props) {
+  return (
+    <Typography
+      color="textSecondary"
+      className={props.selectProps.classes.placeholder}
+      {...props.innerProps}
+    >
+      {props.children}
+    </Typography>
+  )
+}
+
+function SingleValue(props) {
+  return (
+    <Typography
+      className={props.selectProps.classes.singleValue}
+      {...props.innerProps}
+    >
+      {props.children}
+    </Typography>
+  )
+}
+
+function ValueContainer(props) {
+  return (
+    <div className={props.selectProps.classes.valueContainer}>
+      {props.children}
+    </div>
+  )
+}
+
+function MultiValue(props) {
+  return (
+    <Chip
+      tabIndex={-1}
+      label={props.children}
+      className={classNames(props.selectProps.classes.chip, {
+        [props.selectProps.classes.chipFocused]: props.isFocused,
+      })}
+      onDelete={props.removeProps.onClick}
+      deleteIcon={<CancelIcon {...props.removeProps} />}
+    />
+  )
+}
+
+function Menu(props) {
+  return (
+    <Paper
+      square
+      className={props.selectProps.classes.paper}
+      {...props.innerProps}
+    >
+      <MenuList>{props.children}</MenuList>
+    </Paper>
+  )
+}
+
+const components = {
+  Control,
+  Menu,
+  MultiValue,
+  NoOptionsMessage,
+  Option,
+  Placeholder,
+  SingleValue,
+  ValueContainer,
+}
+const ConfigurationsList = ({ user, classes, theme }) => {
+  const [state, setState] = useState({
+    user: {},
+    preferences: {},
+    configurations: [],
+    gridCols: 1,
+    gridWidth: 350,
+    searchUsers: false,
+    friends: [],
+    shared: [],
+    snackTime: false,
+    uploadSnack: false,
+    selectedConfig: {},
+    configOwner: '',
+    totalStudies: 0,
+    totalSubjects: 0,
+    totalDays: 0,
+  })
+
+  useEffect(() => {
+    loadUserNames()
+
+    fetchConfigurations(user.uid).then((configurations) => {
+      setState((prevState) => {
+        return {
+          ...prevState,
+          configurations,
+        }
+      })
+    })
+
+    fetchPreferences(user.uid)
+  }, [])
+
+  useEffect(() => {
+    if (user.message.length > 0) {
+      setState((prevState) => {
+        return {
+          ...prevState,
+          uploadSnack: true,
+        }
+      })
+    }
+    handleResize()
+    window.addEventListener('resize', handleResize)
+    return () => window.removeEventListener('resize', handleResize)
+  }, [user, window, state.gridCols])
```

Comment:
do we want the window and `state.gridCols` in the dependency array? I don't see the `gridCols` used here. Also, is this something we can do on first mount instead of when these attributes change?

Reply:
good catch gridCols can be removed. It's mostly tracking window when it resizes

Reply:
09bf0ca

Reply:
I see the change in the commit but not in this diff. Is that commit part of this PR?